tests/JestProcessManagement/JestProcessManager.test.ts

```diff
@@ -266,6 +264,56 @@ describe('JestProcessManager', () => {
         expect(jestProcessManager.numberOfProcesses).toBe(0)
       })
     })
+    describe('2-staged process: runAllTest => runWatchMode', () => {
+      let startOptions = {}
+      beforeEach(() => {
+        jestProcessMock.mockImplementation(() => {
+          let isStopped = false
+          const stop = () => (isStopped = true)
+          const stopRequested = () => isStopped
+          const onExit = callback => {
+            eventEmitter.on('debuggerProcessExit', callback)
+          }
+          return { onExit, stop, stopRequested }
+        })
+        startOptions = {
+          exitCallback: exitHandler,
+          watchMode: WatchMode.Watch,
+        }
+      })
+      it('normally the watch-mode process will auto-start when first process exit', () => {
+        const p = jestProcessManager.startJestProcess(startOptions)
+        expect(jestProcessMock.mock.instances.length).toBe(1)
+
+        eventEmitter.emit('debuggerProcessExit', p)
+
+        expect(jestProcessMock.mock.instances.length).toBe(2)
+      })
+      it('stopAll will prevent auto start the watch mode process', async () => {
+        const p = jestProcessManager.startJestProcess(startOptions)
+        expect(jestProcessMock.mock.instances.length).toBe(1)
+        expect(jestProcessManager.numberOfProcesses).toBe(1)
+
+        await jestProcessManager.stopAll()
+        expect(jestProcessManager.numberOfProcesses).toBe(0)
+
+        eventEmitter.emit('debuggerProcessExit', p)
+        expect(jestProcessManager.numberOfProcesses).toBe(0)
+        expect(jestProcessMock.mock.instances.length).toBe(1)
+      })
+      it('stopJestProcess will prevent auto start the watch mode process', async () => {
+        const p = jestProcessManager.startJestProcess(startOptions)
+        expect(jestProcessMock.mock.instances.length).toBe(1)
+        expect(jestProcessManager.numberOfProcesses).toBe(1)
+
+        await jestProcessManager.stopJestProcess(p)
+        expect(jestProcessManager.numberOfProcesses).toBe(0)
+
+        eventEmitter.emit('debuggerProcessExit', p)
+        expect(jestProcessManager.numberOfProcesses).toBe(0)
+        expect(jestProcessMock.mock.instances.length).toBe(1)
```

Comment:
It looks like you're asserting that a mock has been called `n` times. How do you feel about using one of the fancy matchers, `.toHaveBeenCalledTimes(number)` or the alias `.toBeCalledTimes`? Here are the [docs](https://jestjs.io/docs/en/expect#tohavebeencalledtimesnumber).